### PR TITLE
SQL: handle X-Pack or X-Pack SQL not being available in a more graceful way

### PR DESCRIPTION
**Pull request by astefan**
Fix for https://github.com/elastic/elasticsearch/issues/30009.
The same error code is received when either the x-pack is not available at all, or SQL is explicitly disabled (`xpack.sql.enabled: false`).

### CODE REVIEW

**Review by matriv — APPROVED** [review]
LGTM

File: x-pack/plugin/sql/sql-client/src/main/java/org/elasticsearch/xpack/sql/client/JreHttpUrlConnection.java [hunk 1]
```diff
@@ -176,6 +176,16 @@ private boolean shouldParseBody(int responseCode) {
         }
         SqlExceptionType type = SqlExceptionType.fromRemoteFailureType(failure.type());
         if (type == null) {
+            if (con.getResponseCode() == HttpURLConnection.HTTP_BAD_REQUEST) {
+                return new ResponseOrException<>(new SQLException("It doesn't look like the X-Pack or the X-Pack SQL component"
```
Comment (costin) [inline x-pack/plugin/sql/sql-client/src/main/java/org/elasticsearch/xpack/sql/client/JreHttpUrlConnection.java]: Slight rephrasing, removing some `the` and typos: `X-Pack/ SQL do not seem to be available on the Elasticsearch node using the access path '...' . Please verify X-Pack is installed and SQL enabled. Alternatively, check if any proxy is interfering the communication to Elasticsearch.`

**Review by costin — APPROVED** [review]
LGTM

File: x-pack/plugin/sql/sql-client/src/main/java/org/elasticsearch/xpack/sql/client/JreHttpUrlConnection.java (hunk 1 above)
Comment (astefan) [reply x-pack/plugin/sql/sql-client/src/main/java/org/elasticsearch/xpack/sql/client/JreHttpUrlConnection.java]: Thanks. I modified it.

**Comment by astefan** [comment]
@costin @matriv sorry to resurrect this one, but the 400 code is returned in other situations as well (looking into a different bug showed the other scenario). I changed the way we check for x-pack/sql being available by comparing the error message itself. Not elegant, but the alternative is probably an additional request being made for every sql request going to ES.

**Pull request merged**